R/paginate.R

```diff
@@ -215,8 +212,31 @@ req_perform_iteratively <- function(req,
   vctrs::list_unchop(out)
 }
 
+#' Retrieve the next request from an iterative response
+#'
+#' In most case you should use [req_perform_iteratively()] but you can
+#' use this lower-level helper if you need more control.
+#'
+#' @inheritParams req_perform
+#' @param parsed The response parsed by the argument `parse_resp` of [req_paginate()].
+#' @keywords internal
 #' @export
-#' @rdname req_perform_iteratively
+#' @return Generates the next request in an iterative request,
+#'   or `NULL` if there are no more pages to return.
+#' @examples
+#' req_flowers <- request(example_url()) %>%
+#'   req_url_path("/iris") %>%
+#'   req_url_query(limit = 40) %>%
+#'   req_paginate_page_index(
+#'     page_index = function(req, page) {
+#'       req %>% req_url_query(page_index = page)
+#'     }
+#'   )
+#' req_flowers %>% .$url
```

Comment:
The pipe seems a bit unnecessary?

```suggestion
#' req_flowers$url
```